core-library/src/main/java/org/silverpeas/core/admin/domain/DomainDriverManager.java

```diff
@@ -192,77 +210,143 @@ public void updateUserFull(UserFull user) throws AdminException {
+      // Get the silverpeas's user information
+      long tmpStart = currentTimeMillis();
+      final List<UserDetail> silverpeasUsers = userDAO.getUserByIds(connection, userIds);
+      if (silverpeasUsers.size() != userIds.size()) {
+        throw new AdminException(failureOnGetting(USERS, userIds));
       }
-
+      fetchingSpUsersTime = currentTimeMillis() - tmpStart;
+      nbSpUsersFetched = silverpeasUsers.size();
       // Get a DomainDriver instance
-      DomainDriver domainDriver = this.getDomainDriver(silverpeasUser.getDomainId());
-
-      // Get User detail from specific domain
-      user = getUserDetail(userId, isUserFull, silverpeasUser, domainDriver);
-
-      // Fill silverpeas info of user details
-      user.setLogin(silverpeasUser.getLogin());
-      user.setId(userId);
-      user.setSpecificId(silverpeasUser.getSpecificId());
-      user.setDomainId(silverpeasUser.getDomainId());
-      user.setAccessLevel(silverpeasUser.getAccessLevel());
-      user.setCreationDate(silverpeasUser.getCreationDate());
-      user.setSaveDate(silverpeasUser.getSaveDate());
-      user.setVersion(silverpeasUser.getVersion());
-      user.setTosAcceptanceDate(silverpeasUser.getTosAcceptanceDate());
-      user.setLastLoginDate(silverpeasUser.getLastLoginDate());
-      user.setNbSuccessfulLoginAttempts(silverpeasUser.getNbSuccessfulLoginAttempts());
-      user.setLastLoginCredentialUpdateDate(silverpeasUser.getLastLoginCredentialUpdateDate());
-      user.setExpirationDate(silverpeasUser.getExpirationDate());
-      user.setState(silverpeasUser.getState());
-      user.setStateSaveDate(silverpeasUser.getStateSaveDate());
-      user.setNotifManualReceiverLimit(silverpeasUser.getNotifManualReceiverLimit());
-
-      if (isUserFull) {
-        user.setLoginQuestion(silverpeasUser.getLoginQuestion());
-        user.setLoginAnswer(silverpeasUser.getLoginAnswer());
+      tmpStart = currentTimeMillis();
+      final Map<String, Pair<String, DomainDriver>> domainsById = new HashMap<>();
+      for(final UserDetail spUser : silverpeasUsers) {
+        final String domainId = spUser.getDomainId();
+        if (!domainsById.containsKey(domainId)) {
+          domainsById.put(domainId, Pair.of(domainId, this.getDomainDriver(domainId)));
+        }
       }
-
+      fetchingDomainTime = currentTimeMillis() - tmpStart;
+      nbDomainsFetched = domainsById.size();
+      tmpStart = currentTimeMillis();
+      final Map<String, T> domainUsersBySpecificId = silverpeasUsers
+          .stream()
+          .collect(groupingBy(u -> domainsById.get(u.getDomainId()), mapping(u -> u, toList())))
+          .entrySet()
+          .stream()
+          // Get User details from specific domain
+          .flatMap(e -> toUsersFromDomain(e.getValue(), isUserFull, e.getKey().getSecond()).stream()
+              .map(u -> {
+                u.setDomainId(e.getKey().getFirst());
+                return u;
+              }))
```

Comment:
But, by passing as argument of the toUsersFromDomain method either the pair with the domain Id and the domain driver (the entry's value of the map) or both the domain id and the domain driver instead of just the second value of the pair (the domain driver) you can have this information in the toUsersFromDomain method

Reply:
Yes, I could have give the pair to the method. It is a question I asked myself. I finally choose to not transmit it to keep in mind that the identifier of a domain is not known by the driver itself.

Reply:
Hum.... and? The method doesn't belong to the domain driver but to the `DomainDriverManager` that can have the knowledge of it. This method requires some arguments to perform its task. For example, it requires the driver manager just to pass it as an argument when constructing a new full user detail (just for that). So, the domain id can be passed to just set the domain id to the user detail. And by doing so, we avoid so an additional browsing of the users, browsing that can be easily avoided here.

Reply:
Yes, the method belong to DomainDriverManager. But as I said I thought about this point, and here is the result.
About the browsing of users, there is 2 ou 3 years ago, I had luck to follow a conference at SnowCamp session which explained to the auditors that it is less costly to perform several browsing to put several same data against to put all the same data into a single browsing. I used also this knowledge here for doing my choice.
Anyway, I see that my reasoning seems not good against yours, so I will change this part of my work in order to be compliant to your exepectings.

Reply:
Modifications have been done.